From ab50e3165e211acbf6ea6f63effa59107607a087 Mon Sep 17 00:00:00 2001
From: Prakash.Kumar <prakumar@habana.ai>
Date: Tue, 08 Jun 2021 06:46:32 +0300
Subject: [PATCH] [SW-31053] greco: arc: support for boot sts error

This patch adds changes to report error in boot
status whenever ACK is not received from ARC 0/1
after reset. Currently print is added to report
the error.

Change-Id: I1ab374846059afae48cb4d7d114108cdad565106
Signed-off-by: Prakash.Kumar <prakumar@habana.ai>
---

diff --git a/arch/greco/common.c b/arch/greco/common.c
index c175dee..9af620a 100644
--- a/arch/greco/common.c
+++ b/arch/greco/common.c
@@ -410,6 +410,7 @@
 	u32 retry = ARC_HALT_ACK_RETRIES;
 	u32 halt_mask, halt_ack_mask;
 	u32 rst_offset;
+	u32 cpu_boot_err_val;
 
 	if (id) {
 		halt_ack_offset = mmPSOC_ARC1_CFG_RUN_HALT_ACK;
@@ -440,6 +441,10 @@
 			retry--;
 			if (!retry) {
 				preboot_puts_hex("No ack from ARC", id, false);
+				cpu_boot_err_val = (!id) ?
+					CPU_BOOT_ERR0_ARC0_HALT_ACK_NOT_RCVD :
+					CPU_BOOT_ERR0_ARC1_HALT_ACK_NOT_RCVD;
+				cpu_boot_err_clrset(cpu_boot_err_val, true);
 				break;
 			}
 		}
@@ -560,6 +565,7 @@
 	u32 run_ack_offset, run_req_offset;
 	u32 run_ack_mask;
 	u64 start;
+	u32 cpu_boot_err_val;
 
 	if (id) {
 		run_ack_offset = mmPSOC_ARC1_CFG_RUN_HALT_ACK;
@@ -582,6 +588,10 @@
 		if (timeout_expired(start, ARC_RUN_ACK_TIMEOUT_US)) {
 			preboot_puts_hex("No running state ack from ARC ",
 					id, false);
+			cpu_boot_err_val = (!id) ?
+				CPU_BOOT_ERR0_ARC0_RUN_ACK_NOT_RCVD :
+				CPU_BOOT_ERR0_ARC1_RUN_ACK_NOT_RCVD;
+			cpu_boot_err_clrset(cpu_boot_err_val, true);
 			break;
 		}
 	}
